From 7b781d1a67486a2ae50c174731ca1daebf381ab4 Mon Sep 17 00:00:00 2001
From: Fabio Estevam <festevam@gmail.com>
Date: Fri, 23 Mar 2012 13:13:49 -0300
Subject: [PATCH] ARM: mx28: Remove CPUFREQ support

When using CPUFREQ driver with the mainline U-boot, the kernel hangs.

Remove CPUFREQ support for now.

Signed-off-by: Fabio Estevam <fabio.estevam@freescale.com>
---
 arch/arm/configs/imx28evk_defconfig |    1 -
 arch/arm/plat-mxs/Kconfig           |    2 --
 arch/arm/plat-mxs/clock.c           |   10 ----------
 3 files changed, 0 insertions(+), 13 deletions(-)

diff --git a/arch/arm/configs/imx28evk_defconfig b/arch/arm/configs/imx28evk_defconfig
index 08f6a18..0c8bc66 100644
--- a/arch/arm/configs/imx28evk_defconfig
+++ b/arch/arm/configs/imx28evk_defconfig
@@ -18,7 +18,6 @@ CONFIG_TRACE_IRQFLAGS_SUPPORT=y
 CONFIG_HARDIRQS_SW_RESEND=y
 CONFIG_GENERIC_IRQ_PROBE=y
 CONFIG_RWSEM_GENERIC_SPINLOCK=y
-CONFIG_ARCH_HAS_CPUFREQ=y
 CONFIG_GENERIC_HWEIGHT=y
 CONFIG_GENERIC_CALIBRATE_DELAY=y
 CONFIG_ZONE_DMA=y
diff --git a/arch/arm/plat-mxs/Kconfig b/arch/arm/plat-mxs/Kconfig
index 831a24b..63768f8 100644
--- a/arch/arm/plat-mxs/Kconfig
+++ b/arch/arm/plat-mxs/Kconfig
@@ -13,7 +13,6 @@ config ARCH_MX28
 	select MXS_ICOLL
 	select MXS_DMA_ENGINE
 	select MXS_AUART_DMA_SUPPORT
-	select ARCH_HAS_CPUFREQ
 	---help---
 	 Support Freescale MX28 chips
 
@@ -25,7 +24,6 @@ config ARCH_MX23
 	select MXS_ICOLL
 	select MXS_DMA_ENGINE
 	select MXS_AUART_DMA_SUPPORT
-	select ARCH_HAS_CPUFREQ
 	---help---
 	Support Freescale MX23 chips
 
diff --git a/arch/arm/plat-mxs/clock.c b/arch/arm/plat-mxs/clock.c
index 1b98b1e..3ce7953 100644
--- a/arch/arm/plat-mxs/clock.c
+++ b/arch/arm/plat-mxs/clock.c
@@ -119,11 +119,6 @@ int clk_enable(struct clk *clk)
 
 	ret = __clk_enable(clk);
 	spin_unlock_irqrestore(&clockfw_lock, flags);
-	if ((clk->flags & CPU_FREQ_TRIG_UPDATE)
-	    && (pre_usage == 0)) {
-		cpufreq_trig_needed = 1;
-		cpufreq_update_policy(0);
-	}
 	return ret;
 }
 EXPORT_SYMBOL(clk_enable);
@@ -139,11 +134,6 @@ void clk_disable(struct clk *clk)
 	spin_lock_irqsave(&clockfw_lock, flags);
 	__clk_disable(clk);
 	spin_unlock_irqrestore(&clockfw_lock, flags);
-	if ((clk->flags & CPU_FREQ_TRIG_UPDATE)
-			&& ((clk->ref & CLK_EN_MASK) == 0)) {
-		cpufreq_trig_needed = 1;
-		cpufreq_update_policy(0);
-	}
 }
 EXPORT_SYMBOL(clk_disable);
 
-- 
1.7.1

